src/sql/azdata.d.ts

```diff
@@ -0,0 +1,37 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+declare module 'azdata' {
+	import * as vscode from 'vscode';
+
+	/**
+	 * Connection profile primary class
+	 */
+	export class ConnectionProfile {
```

Comment:
We've previously used interfaces (instead of classes) since they are build-time only and don't require adding additional class implementations to the extension host.  If we'd like to provide helper methods such as createFrom then we'll need to switch to classes.